File: 0495.提莫攻击/0495-提莫攻击.cpp
class Solution {
public:
    int findPoisonedDuration(vector<int>& timeSeries, int duration) {
        int ans=0;
        int expired=0;
        int n=timeSeries.size();
        for (int i=0;i<n;i++){
            if(timeSeries[i]>=expired){
                ans+=duration;
            }
            else{
                ans+=timeSeries[i]+duration-expired;
            }
        expired=timeSeries[i]+duration;
        }
        return ans;
    }
};

File: 0495.提莫攻击/0495-提莫攻击.py
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        ans,not_du=0,0
        for i in range(len(timeSeries)):
            if timeSeries[i]>=not_du:
                ans+=duration
            else:
                ans+=timeSeries[i]+duration-not_du
            not_du=timeSeries[i]+duration
        return ans